fairlearn/utils/_input_validation.py

```diff
@@ -69,7 +69,7 @@ def _validate_and_reformat_input(X, y=None, expect_y=True, enforce_binary_labels
         elif isinstance(y, pd.DataFrame) and y.shape[1] == 1:
             y = y.to_numpy().reshape(-1)
 
-        X, y = check_X_y(X, y)
+        X, y = check_X_y(X, y, dtype=None, force_all_finite=False)
         y = check_array(y, ensure_2d=False, dtype='numeric')
```

Comment:
does `y` really need to be `numeric`? Or is it also something the underlying estimator should check? I think for a classification `y` could be categorical?

Reply:
I believe that the way the reweighting currently works, it has to be 0 or 1. I agree that it ought to be possible to take any two categorical values, but that would be a bigger change.